core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -4642,13 +4642,18 @@ private FileSystemMasterAuditContext createAuditContext(String command, AlluxioU
         new FileSystemMasterAuditContext(auditLogWriter);
     if (auditLogWriter != null) {
       String user = null;
+      String connectionUser = null;
       String ugi = "";
       try {
         user = AuthenticatedClientUser.getClientUser(ServerConfiguration.global());
+        connectionUser = AuthenticatedClientUser.getConnectionUser(ServerConfiguration.global());
       } catch (AccessControlException e) {
         ugi = "N/A";
       }
       if (user != null) {
+        if (connectionUser != null && !connectionUser.equals(user)) {
+          user = connectionUser + " (auth=PROXY) via " + user;
+        }
         try {
           String primaryGroup = CommonUtils.getPrimaryGroupName(user, ServerConfiguration.global());
```

Comment:
I'm a little bit concerned about whether or not you will get the expected group from `CommonUtils.getPrimaryGroupName` when you pass in `connectionUser + " (auth=PROXY) via " + user`.

Reply:
Thanks for your review, i'll fix it like this:
```java
      if (user != null) {
        try {
          String primaryGroup = CommonUtils.getPrimaryGroupName(user, ServerConfiguration.global());
          ugi = user + "," + primaryGroup;
        } catch (IOException e) {
          LOG.debug("Failed to get primary group for user {}.", user);
          ugi = user + ",N/A";
        }
        if (connectionUser != null && !connectionUser.equals(user)) {
          ugi = connectionUser + " (auth=PROXY) via " + ugi;
        }
      }
```